src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1981,8 +1981,16 @@ def get_multisite_status(self):
             is_multisite_configured = False
         return is_multisite_configured
 
-    def get_multisite_sync_status(self):
+    def get_multisite_sync_status(self, daemon_name: str):
+        if daemon_name:
+            orch = OrchClient.instance()
+            daemons = [d.to_dict() for d in orch.services.list_daemons()]
+            rgw_daemons = [d for d in daemons if d['daemon_type'] == 'rgw']  # noqa E501 #pylint: disable=line-too-long
+            active_rgw_daemon = [d for d in rgw_daemons if d['daemon_id'] == daemon_name]
+            realm_name = active_rgw_daemon['realm_name']
```

Comment:
Can we move this code to another function, `_get_active_daemon()`, following the single responsibility principle, functions should aim to only do one thing.

Reply:
Done, Thanks